### ISSUE TITLE
v4.21.0 : broken highlight code ?

### ISSUE THREAD

**Issue by bthieurmel** [almende/vis#3589]
Hi,

It seems that new v4.21.0 breaks  neighbourhood Highlight example : 

http://visjs.org/examples/network/exampleApplications/neighbourhoodHighlight.html

**Comment by wimrijnders** [comment]
Just checked it, working as intended. What are you seeing to conclude that it is broken?

Maybe it's a platform thing. I tested on linux+firefox/chrome and it looks good. What's your OS+browser?

**Comment by bthieurmel** [comment]
Nodes and edges don't became gray on click.

windows + google chrome

**Comment by wimrijnders** [comment]
Ah, a color thing. I'm the worst possible guy to examine it, then 😄.

Yes, it does say that in the description. But I see it's working because the edges in question turn bold. 

Okay, I'll have a look. Maybe the description needs to be changed instead (whatever it takes to make it consistent).

**Comment by bthieurmel** [comment]
Thanks. 

I'm the developper of a ``R`` package ``visNetwork`` based on the great library ``vis.js``.  

And I've some special options based on this kind of functionnality (change colors of subset of nodes / edges on click), and so all are broken with the current ``4.21.0``...!

(If you want to have a look : http://datastorm-open.github.io/visNetwork/options.html)

**Comment by wimrijnders** [comment]
Ah, `R`, rings a bell. We've interacted via issues before.

I followed the link (and the page is a candidate for showcasing IMHO), but it doesn't help much. Would it be possible to isolate the relevant options in an example?

**Comment by bthieurmel** [comment]
Yes, I redirected some issue send on ``visNetwork`` to ``vis.js``....

The link was just for informations, not to help for this issue....(and the site is still under ``4.20``). So, I think the neighbourhood Highlight example is a good case to try to fix that ?

http://visjs.org/examples/network/exampleApplications/neighbourhoodHighlight.html

**Comment by wimrijnders** [comment]
OK, on my TODO. Don't want to do it right now due to tax people becoming increasingly agitated by me, on the grounds of me not having my paperwork done.

**Comment by bthieurmel** [comment]
First observation : seems to be broken if node have a ``group`` value.

**Comment by wimrijnders** [comment]
I'm ashamed to say that it took me a while to understand the issue; the highlight is set properly, but in previous versions **all the other nodes and edges were grayed out**. And that is not happening in latest release.

What is happening now, is that the groups override the color values, so the gray color never appears. I will have to treat the colors in the same way as the fonts are treated, where the order is now strictly forced.

I guess this means that this issue is confirmed.

**Comment by bthieurmel** [comment]
Not only a color problem, but the issue is for all parameters set in groups and in nodes.

**Comment by wimrijnders** [comment]
Except for fonts! I can vouch for that. I fixed that particular issue: #3486. Easily the toughest thing I had to fix within `Network`. I guess you can view this as fallout.

**Comment by bthieurmel** [comment]
``vis.js`` documentation confirme that individual nodes parameters must be prioritary to group options: 

``Node specific styling overrides group styling.``

**Comment by wimrijnders** [comment]
And that's how it should be. I'm not disputing that.

This just confirms for me that the options handling is broken.

**Comment by bthieurmel** [comment]
Thanks for quick answers, Good luck and thank you for your work...!

**Comment by wimrijnders** [comment]
OK. I personally guarantee that this will get into the bugfix release (planned weekend of 28 nov). For me, this is a matter of having enough coffee. If this issue is hindering you, I think the best thing is to go back to the previous version for the time being.

**Comment by david-bc** [comment]
Confirmed.

Falling back to `npm i -S vis@4.20.1 --save-exact` solves the group/color issue

**Comment by shaildegoam** [comment]
It is still not working :(
